internal/subscription/core.go

```diff
@@ -103,9 +101,37 @@ func (c *Core) CreateSubscription(ctx context.Context, m *Model) error {
 		})
 
 		if err != nil {
-			logger.Ctx(ctx).Errorw("failed to create deadletter topic for subscription", "name", m.GetDeadLetterTopic(), "error", err.Error())
+			logger.Ctx(ctx).Errorw("failed to create dead letter topic for subscription", "name", m.GetDeadLetterTopic(), "error", err.Error())
 			return err
 		}
+
+		// this creates the needed retry/delay topic configs
+		m.DelayConfig, err = NewDelayConfig(m)
```

Comment:
do we need a separate delay config in subscription model, we have retry/dlq configs already, most of the delayconfig is just copy of same

Reply:
This was created to keep track of the delay-topics created for a subscripition.